mcodingbot/utils/plugin.py

```diff
@@ -0,0 +1,16 @@
+from __future__ import annotations
+
+import typing
+
+import crescent
+
+if typing.TYPE_CHECKING:
+    from mcodingbot.bot import Bot
+
+__all__: typing.Sequence[str] = ("Plugin",)
+
+
+class Plugin(crescent.Plugin):
+    @property
+    def app(self) -> Bot:
+        return typing.cast("Bot", super().app)
```

Comment:
Why is this using `super()` and not self?

Reply:
pretty sure `self.app` would cause infinite recursion

Reply:
oh, recursion - i forgot when i did this i used bot and app properties to get around that.